### PR TITLE
Add support for user to input multiple dateTime formats

### LINKED ISSUE #71: As a new user, I can input different DateTime formats in my transactions
...so that I can use the application with ease.

DateTime formats supported: dd-MM-yyyy HH:mm, yyyy-MM-dd HH:mm, dd MMM yyyy HH:mm

### PR DESCRIPTION
**Pull request by jamz903**
Closes #71 

Adds support for 3 dateTime formats that can be input by the user (instead of the initial 1)
1. dd-MM-yyyy HH:mm
2. yyyy-MM-dd HH:mm
3. dd MMM yyyy HH:mm

Stores dateTime in the format that the user inputs (any of the above 3).

### CODE REVIEW

File: src/test/java/unicash/model/transaction/DateTimeTest.java [hunk 1]
```diff
@@ -1,8 +1,10 @@
 package unicash.model.transaction;
```
Comment (elhy1999) [inline src/test/java/unicash/model/transaction/DateTimeTest.java:2]: Now that we have three different ways to input datetimes, how shall we define equality between our DateTime objects? If we only take the `dateTime` attribute into consideration, then we could consider adding unit tests to check if dateTime objects that have been instantiated in different ways are still equal. That is:
```
DateTime d1 = new DateTime("01-01-2023 00:00");
DateTime d2 = new DateTime("2023-01-01 00:00");
assertTrue(d1.equals(d2));
```

If we also want to compare the value of `originalString`, then we will need to bake that logic into the DateTime class's `equals` method! 😄

**Review by elhy1999 — DISMISSED** [review]
Thanks for the hard work! Overall, it LGTM, but I added a few pointers you can consider changing before merging 😄

File: src/test/java/unicash/model/transaction/DateTimeTest.java (hunk 1 above)
Comment (jamz903) [reply src/test/java/unicash/model/transaction/DateTimeTest.java:2]: My understanding of this is that regardless of the input format by the user, the DateTime would be equal as long as it is the same date & time. I'll ask the team about this during our meeting, to double confirm. Thanks for the suggestion for the unit tests, i'll add that in!

File: src/main/java/unicash/model/transaction/DateTime.java [hunk 2]
```diff
@@ -14,11 +15,17 @@
...
     public static final String MESSAGE_CONSTRAINTS =
-            "DateTime should be in the following format " + DATETIME_PATTERN;
-    public static final DateTimeFormatter DATETIME_FORMATTER = DateTimeFormatter
-            .ofPattern(DATETIME_PATTERN);
+            "DateTime should be in either of the following formats: "
+                    + "dd-MM-yyyy HH:mm, yyyy-MM-dd HH:mm, dd MMM yyyy HH:mm";
+    //accept date in multiple formats
+    private static final DateTimeFormatterBuilder DATETIME_FORMATTER_BUILDER =
+            new DateTimeFormatterBuilder().append(DateTimeFormatter.ofPattern("[dd-MM-uuuu HH:mm]"
+                    + "[uuuu-MM-dd HH:mm]" + "[dd MMM uuuu HH:mm]"));
```
Comment (woojiahao) [inline src/main/java/unicash/model/transaction/DateTime.java:25]: Maybe split this across lines?

**Review by woojiahao — CHANGES_REQUESTED** [review]
LGTM, good work, just minor comment

File: src/main/java/unicash/model/transaction/DateTime.java [hunk 3]
```diff
@@ -66,13 +80,13 @@ public LocalDateTime getDateTime() {
...
-     * Helper method to stringify LocalDateTime objects into original text string
-     * pass by the user.
+     * Helper method to return original text string
+     * passed by the user.
      *
      * @return text string of the LocalDateTime object
      */
     public String originalString() {
-        return dateTime.format(DateTimeFormatter.ofPattern(DATETIME_PATTERN));
+        return originalDateTime;
```
Comment (woojiahao) [inline src/main/java/unicash/model/transaction/DateTime.java:89]: Inline with the comment in #96 where `originalString` might seen misleading, maybe we can rename this to `inputString` or something if we're looking for uniformity in naming

**Review by woojiahao — CHANGES_REQUESTED** [review]
Minor nit in the `originalString` name but other than that, LGTM

**Review by woojiahao — APPROVED** [review]
LGTM, thanks for the quick turnaround!

**Pull request merged**